iml-wasm-components/iml-input/src/lib.rs

```diff
@@ -0,0 +1,118 @@
+extern crate combine;
+extern crate seed;
+
+use combine::stream::easy::{Error, Errors, Info};
+use seed::{div, input, prelude::*};
+
+use iml_tooltip::TooltipPlacement;
+
+pub struct Model<T> {
+    pub value: Result<T, String>,
+    pub tooltip: iml_tooltip::Model,
+}
+
+impl<T> Default for Model<T> {
+    fn default() -> Self {
+        Self {
+            value: Err(String::new()),
```

Comment:
Do we want the default state to be an empty error?

Reply:
I think it could be the default of underlying type.

Reply:
I suppose this means the component starts in the Error state?

Reply:
Anyway, `Ok` does not seem right, because it is not ok - we don't know yet the result. So `Err` looks the best. Then, I don't want to show  a user an error tooltip while they hasn't typed anything yet. So empty string indicates no user error, but still invalid state of input. I guess I can user `Option` instead.